src/services/identity/postgres.ts

```diff
@@ -232,36 +240,69 @@ export class PostgresIdentityService extends DefaultIdentityService {
 
 	async createStatusList2021(did: string, resourceOptions: ResourcePayload, statusOptions: CreateStatusListOptions, agentId: string): Promise<CreateStatusList2021Result> {
 		const agent = await this.createAgent(agentId)
+        if (!await CustomerService.instance.find(agentId, { did })) {
+            throw new Error(`${did} not found in wallet`)
+        }
 		return await Veramo.instance.createStatusList2021(agent, did, resourceOptions, statusOptions)
 	}
 
 	async updateStatusList2021(did: string, statusOptions: UpdateStatusListOptions, publish: boolean, agentId: string): Promise<BulkRevocationResult | BulkSuspensionResult | BulkUnsuspensionResult> {
 		const agent = await this.createAgent(agentId)
+        if (!await CustomerService.instance.find(agentId, { did })) {
+            throw new Error(`${did} not found in wallet`)
+        }
 		return await Veramo.instance.updateStatusList2021(agent, did, statusOptions, publish)
 	}
 
 	async checkStatusList2021(did: string, statusOptions: CheckStatusListOptions, agentId: string): Promise<StatusCheckResult> {
 		const agent = await this.createAgent(agentId)
+        if (!await CustomerService.instance.find(agentId, { did })) {
+            throw new Error(`${did} not found in wallet`)
+        }
 		return await Veramo.instance.checkStatusList2021(agent, did, statusOptions)
 	}
 
 	async broadcastStatusList2021(did: string, resourceOptions: ResourcePayload, statusOptions: BroadCastStatusListOptions, agentId: string): Promise<boolean> {
 		const agent = await this.createAgent(agentId)
+        if (!await CustomerService.instance.find(agentId, { did })) {
+            throw new Error(`${did} not found in wallet`)
+        }
 		return await Veramo.instance.broadcastStatusList2021(agent, did, resourceOptions, statusOptions)
 	}
 
 	async revokeCredentials(credentials: VerifiableCredential | VerifiableCredential[], publish: boolean, agentId: string) {
 		const agent = await this.createAgent(agentId)
+        for(var credential in credentials) {
```

Comment:
If it's a common functionality let's make it as a function, maybe helper

Reply:
made it as a function within the same class as it's not gonna be used in others